npipe_windows.go

```diff
@@ -443,6 +446,12 @@ func (c *PipeConn) Write(b []byte) (int, error) {
 
 // Close closes the connection.
 func (c *PipeConn) Close() error {
+	c.closedMutex.Lock()
+	defer c.closedMutex.Unlock()
+	if c.closed {
+		return nil
+	}
+	c.closed = true
 	return syscall.CloseHandle(c.handle)
```

Comment:
Strictly speaking, we're unnecessarily holding the mutex while calling out to CloseHandle. An atomic compare/exchange would be better.